pkg/virt-controller/watch/vm.go

```diff
@@ -3297,17 +3297,13 @@ func (c *VMController) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi
 
 	conditionManager := controller.NewVirtualMachineInstanceConditionManager()
 
-	if vmi.Spec.Domain.Memory.MaxGuest == nil ||
-		conditionManager.HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceMemoryChange, k8score.ConditionFalse) {
-		// memory hotplug either failed or is incompatible, let's trigger the restart required condition
-		// to make the memory bump effective
-
+	if conditionManager.HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceMemoryChange, k8score.ConditionFalse) {
 		vmConditions := controller.NewVirtualMachineConditionManager()
 		vmConditions.UpdateCondition(vm, &virtv1.VirtualMachineCondition{
 			Type:               virtv1.VirtualMachineRestartRequired,
 			LastTransitionTime: metav1.Now(),
 			Status:             k8score.ConditionTrue,
```

Comment:
Does this still remain true?

Reply:
sorry Alice can you elaborate a bit? What does still remain true?

Reply:
If the change cannot be applied for any reason, I would put the status of the condition to false.

Reply:
We set the condition when a live update cannot be applied.